src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,749 @@
+
+    Args:
+        tf_dfs (dict): A dictionary of (partially) transformed dataframes,
+            to be cleaned up.
+
+    Returns:
+        dict: The input dictionary of dataframes, but with a finished
+        respondent_id_ferc714 dataframe.
+
+    """
+    df = (
+        tf_dfs["respondent_id_ferc714"].assign(
+            utility_name_ferc714=lambda x: x.utility_name_ferc714.str.strip(),
+            utility_id_eia=lambda x: x.utility_id_eia.replace(
+                to_replace=0, value=pd.NA)
+        )
+        # These excludes fake Test IDs -- not real planning areas
+        .query("utility_id_ferc714 not in @BAD_RESPONDENTS")
+    )
+    # There are a few utilities that seem mappable, but missing:
+    for rid in MISSING_UTILITY_ID_EIA:
+        df.loc[df.utility_id_ferc714 == rid,
+               "utility_id_eia"] = MISSING_UTILITY_ID_EIA[rid]
+    tf_dfs["respondent_id_ferc714"] = df
+    return tf_dfs
+
+
+def pa_demand_hourly(tf_dfs):
+    """
+    Transform the hourly demand time series by Planning Area.
+
+    Args:
+        tf_dfs (dict): A dictionary of (partially) transformed dataframes,
+            to be cleaned up.
+
+    Returns:
+        dict: The input dictionary of dataframes, but with a finished
+        pa_demand_hourly_ferc714 dataframe.
+
+    """
+    logger.info("Converting dates into pandas Datetime types.")
+    df = (
+        tf_dfs["pa_demand_hourly_ferc714"].assign(
+            report_date=lambda x: pd.to_datetime(x.report_date),
+            utc_offset_code=lambda x: x.utc_offset_code.str.upper().str.strip(),
+        )
+    )
+
+    logger.info("Melting daily FERC 714 records into hourly records.")
+    df = (
+        df.rename(columns=_hours_to_ints)
+        .melt(
+            id_vars=["report_year", "utility_id_ferc714",
+                     "report_date", "utc_offset_code"],
+            var_name="hour",
+            value_name="demand_mwh")
+    )
+    _log_dupes(df, ["utility_id_ferc714", "report_date", "hour"])
+
+    df = (
+        df.pipe(_standardize_offset_codes, tz_fixes=TZ_FIXES)
+        # Discard records lacking *both* UTC offset code and non-zero demand
+        # In practice, this should be *all* of the XXX records, but we're being
+        # conservative, in case something changes / goes wrong. We want to
+        # notice if for some reason later we find XXX records that *do* have
+        # real demand associated with them.
+        .query("utc_offset_code!='XXX' | demand_mwh!=0.0")
+        # Almost all 25th hours are unusable (0.0 or daily totals),
+        # and they shouldn't really exist at all based on FERC instructions.
+        .query("hour!=25")
+        # Switch to using 0-23 hour notation, and combine hour with date:
+        .assign(
+            local_time=lambda x:
+                x.report_date + pd.to_timedelta(x.hour - 1, unit="h")
+        )
+    )
+
+    # Set a few year-specific offset codes:
+    df.loc[(df.report_year == 2006) & (
```

Comment:
this might be more clear or easy to see if these problem children are enumerated in a global variable up top and dealt with in a standardized way down here.

Reply:
Okay I've done this for the by-year-and-utility fixes.  There are only two cases where the demand was reported with a sign error so for now, I've just left that inline.